iroh/src/downloader/get.rs

```diff
@@ -0,0 +1,561 @@
+                            FailureAction::RetryLater(_) => inc!(Metrics, downloads_notfound),
+                            _ => inc!(Metrics, downloads_error),
+                        }
+                    }
+                    Err(e)
+                }
+            }
+        };
+        fut.boxed_local()
+    }
+}
+
+impl From<quinn::ConnectionError> for FailureAction {
+    fn from(value: quinn::ConnectionError) -> Self {
+        // explicit match just to be sure we ar taking everything into account
+        match value {
+            e @ quinn::ConnectionError::VersionMismatch => {
+                // > The peer doesn't implement any supported version
+                // unsupported version is likely a long time error, so this peer is not usable
+                FailureAction::DropPeer(e.into())
+            }
+            e @ quinn::ConnectionError::TransportError(_) => {
+                // > The peer violated the QUIC specification as understood by this implementation
+                // bad peer we don't want to keep around
+                FailureAction::DropPeer(e.into())
+            }
+            e @ quinn::ConnectionError::ConnectionClosed(_) => {
+                // > The peer's QUIC stack aborted the connection automatically
+                // peer might be disconnecting or otherwise unavailable, drop it
+                FailureAction::DropPeer(e.into())
+            }
+            e @ quinn::ConnectionError::ApplicationClosed(_) => {
+                // > The peer closed the connection
+                // peer might be disconnecting or otherwise unavailable, drop it
+                FailureAction::DropPeer(e.into())
+            }
+            e @ quinn::ConnectionError::Reset => {
+                // > The peer is unable to continue processing this connection, usually due to having restarted
+                // TODO(@divma): peer is unavailable but might be available later, maybe retry?
+                FailureAction::DropPeer(e.into())
+            }
+            e @ quinn::ConnectionError::TimedOut => {
+                // > Communication with the peer has lapsed for longer than the negotiated idle timeout
+                // TODO(@divma): my understanding is that quinn should be configured to ping often
+                // enough to prevent this. If the peer's connfiguration and our configuration allow
+                // this to happen maybe the peer is not really usable
+                FailureAction::DropPeer(e.into())
+            }
+            e @ quinn::ConnectionError::LocallyClosed => {
+                // > The local application closed the connection
+                // TODO(@divma): don't see how this is reachable but let's just not use the peer
+                FailureAction::DropPeer(e.into())
+            }
+        }
+    }
+}
+
+impl From<quinn::ReadError> for FailureAction {
+    fn from(value: quinn::ReadError) -> Self {
+        match value {
+            quinn::ReadError::Reset(_)
```

Comment:
can't we retry here, for the same reasons as above?

Reply:
done in 4a7e4b2e